File: test_database.py
#!/usr/bin/env python3
"""
Test script to verify database integration
"""

import asyncio
import sys
import os

sys.path.append(os.path.join(os.path.dirname(__file__), 'lib'))

async def test_database():
    """Test the database functions"""
    try:
        from database import db_manager
        
        print("=== Testing Database Integration ===")
        
        # Initialize database
        await db_manager.initialize()
        await db_manager.insert_sample_data()
        
        # Test the database functions
        print("\n1. Testing get_services()...")
        services = await db_manager.get_services()
        print(f"Found {len(services)} services:")
        for service in services:
            print(f"  - {service['name']}: {service['description']}")
        
        print("\n2. Testing get_countries_for_service('Patent Application')...")
        countries = await db_manager.get_countries_for_service("Patent Application")
        print(f"Found {len(countries)} countries for Patent Application:")
        for country in countries:
            print(f"  - {country['name']} ({country['currency']})")
        
        print("\n3. Testing get_items_for_service_country('Patent Application', 'Malaysia')...")
        items = await db_manager.get_items_for_service_country("Patent Application", "Malaysia")
        print(f"Found {len(items)} items for Patent Application in Malaysia:")
        for item in items:
            print(f"  - {item['name']}: {item['currency']} {item['total_cost']:,.2f}")
        
        print("\n4. Testing generate_quotation_summary...")
        quotation = await db_manager.generate_quotation_summary("Patent Application", "Malaysia", "Basic Patent Application")
        if quotation:
            print("Quotation Summary:")
            print(f"  Service: {quotation['service']}")
            print(f"  Country: {quotation['country']}")
            print(f"  Item: {quotation['item']['name']}")
            print(f"  Total Cost: {quotation['item']['currency']} {quotation['item']['total_cost']:,.2f}")
            print(f"  Quotation ID: {quotation['quotation_id']}")
        else:
            print("Failed to generate quotation summary")
        
        await db_manager.close()
        print("\n=== Database test completed successfully! ===")
        
    except Exception as e:
        print(f"Error testing database: {e}")
        import traceback
        traceback.print_exc()

if __name__ == "__main__":
    asyncio.run(test_database())

File: test-database.ts
// Test script for TypeScript database service
import { databaseService } from './lib/database-service'
import dotenv from 'dotenv'

// Load environment variables
dotenv.config()

async function testDatabaseService() {
  try {
    console.log('Testing TypeScript Database Service...')
    
    // Initialize database
    await databaseService.initialize()
    console.log('✅ Database initialized successfully')
    
    // Test getServices
    console.log('\n📋 Testing getServices...')
    const services = await databaseService.getServices()
    console.log('Services found:', services.length)
    services.forEach(service => {
      console.log(`  - ${service.name}: ${service.description}`)
    })
    
    // Test getCountriesForService
    console.log('\n🌍 Testing getCountriesForService...')
    const copyrightCountries = await databaseService.getCountriesForService('COPYRIGHT')
    console.log('Countries for COPYRIGHT:', copyrightCountries.length)
    copyrightCountries.forEach(country => {
      console.log(`  - ${country.name} (${country.currency})`)
    })
    
    // Test getItemsForServiceCountry
    console.log('\n📦 Testing getItemsForServiceCountry...')
    const laosItems = await databaseService.getItemsForServiceCountry('COPYRIGHT', 'LAOS')
    console.log('Items for COPYRIGHT in LAOS:', laosItems.length)
    laosItems.forEach(item => {
      console.log(`  - ${item.name}`)
      console.log(`    Professional Fee: ${item.professional_fee} ${item.currency}`)
      console.log(`    Official Fee: ${item.official_fee} ${item.currency}`)
      console.log(`    Disbursement: ${item.disbursement} ${item.currency}`)
      console.log(`    Total Cost: ${item.total_cost} ${item.currency}`)
    })
    
    // Test generateQuotationSummary
    console.log('\n📄 Testing generateQuotationSummary...')
    const quotation = await databaseService.generateQuotationSummary(
      'COPYRIGHT', 
      'LAOS', 
      'OFFICIAL COPYRIGHT RECORDATION - FILING NOTIFICATION OF COPYRIGHT AND DESCRIPTION OF COPYRIGHTED WORK WITH LAOS COPYRIGHT OFFICE'
    )
    console.log('Generated quotation:')
    console.log(`  Service: ${quotation.service}`)
    console.log(`  Country: ${quotation.country}`)
    console.log(`  Item: ${quotation.item.name}`)
    console.log(`  Quotation ID: ${quotation.quotation_id}`)
    console.log(`  Generated At: ${quotation.generated_at}`)
    console.log(`  Valid Until: ${quotation.valid_until}`)
    
    // Test getAllRecordsForService
    console.log('\n📊 Testing getAllRecordsForService...')
    const allCopyrightRecords = await databaseService.getAllRecordsForService('COPYRIGHT')
    console.log('All COPYRIGHT records:', allCopyrightRecords.length)
    allCopyrightRecords.forEach(record => {
      console.log(`  - ${record.service} | ${record.country} | ${record.item}`)
      console.log(`    Fees: ${record.prof_fee} + ${record.official_fee} + ${record.disbursement} = ${record.prof_fee + record.official_fee + record.disbursement} ${record.currency}`)
    })
    
    console.log('\n✅ All database tests completed successfully!')
    
  } catch (error) {
    console.error('❌ Database test failed:', error)
    process.exit(1)
  } finally {
    // Close database connection
    await databaseService.close()
    console.log('🔌 Database connection closed')
  }
}

// Run the test
testDatabaseService()
